devops/infrastructure-as-code/aws/cli-prefect2-ecs-agent/deploy_ecs.sh

```diff
@@ -0,0 +1,203 @@
+# if your flow needs access to other AWS resources other than S3, add those in the task role policy: line 96-108
+export AWS_REGION=us-east-2
+export ECS_CLUSTER_NAME=orionEcsCluster
+export ECS_LOG_GROUP_NAME=/ecs/orionEcsAgent
+export ECS_SERVICE_NAME=orionECSAgent
+export PREFECT_SERVICE_ACCOUNT_KEY=""
+export WORK_QUEUE_NAME=default
+export ACCOUNT_ID=""
+export WORKSPACE_ID=""
+export AWS_PAGER=""
+
+# adjust capacity providers to your needs
+aws ecs create-cluster --cluster-name $ECS_CLUSTER_NAME --region $AWS_REGION
+
+cat > ecs_tasks_trust_policy.json<<EOL
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": [
+          "ecs-tasks.amazonaws.com"
+        ]
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOL
+
+aws iam create-role --role-name orionECSAgentTaskExecutionRole \
+--assume-role-policy-document file://ecs_tasks_trust_policy.json --region $AWS_REGION
+
+aws iam attach-role-policy --role-name orionECSAgentTaskExecutionRole \
+--policy-arn "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+
+cat > ecs_tasks_execution_role.json<<EOL
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ssm:GetParameters"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+EOL
+
+aws iam put-role-policy --role-name orionECSAgentTaskExecutionRole --policy-name orionECSAgentTaskExecutionRolePolicy --policy-document file://ecs_tasks_execution_role.json
+
+
+# permissions needed by orion to register new task definitions, deregister old ones, and create new flow runs as ECS tasks
+cat > ecs_task_role.json<<EOL
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ec2:AuthorizeSecurityGroupIngress",
+                "ec2:CreateSecurityGroup",
+                "ec2:CreateTags",
+                "ec2:DescribeNetworkInterfaces",
+                "ec2:DescribeSecurityGroups",
+                "ec2:DescribeSubnets",
+                "ec2:DescribeVpcs",
+                "ec2:DeleteSecurityGroup",
+                "ecs:CreateCluster",
+                "ecs:DeleteCluster",
+                "ecs:DeregisterTaskDefinition",
+                "ecs:DescribeClusters",
+                "ecs:DescribeTaskDefinition",
+                "ecs:DescribeTasks",
+                "ecs:ListAccountSettings",
+                "ecs:ListClusters",
+                "ecs:ListTaskDefinitions",
+                "ecs:RegisterTaskDefinition",
+                "ecs:RunTask",
+                "ecs:StopTask",
+                "iam:PassRole",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents",
+                "logs:DescribeLogGroups",
+                "logs:GetLogEvents",
+                "codecommit:PutFile",
+                "codecommit:GetObjectIdentifier",
```

Comment:
Everything in here except for codecommit I think fits, codecommit access is not required by the ECS agent. Can we remove the codecommit permissions?

Reply:
> This recipe is for the CLI, so I'd like to keep the readme based, procedural feel. I like the additional explicit policy additions but I'd like to split them out from the shell script. @prefectcboyd I'd like your thoughts here.

Agreed , and I think you both have valid points -  keeping the original Readme as a CLI based procedural doc.
Then moving this work as a sub-folder to script the process, with a separate readme, to describe usage of this.
Does that make sense?

The net effect would be :

##Current 
cli-prefect2-ecs-agent/
- Readme.md
- prefect-agent-td.json
- trust-policy.json

##Suggested
cli-prefect2-ecs-agent/
- Readme.md
- prefect-agent-td.json
- trust-policy.json
- scripted_deploy/
   - Readme.md
   - deploy_ecs.sh
   - ecs_task_role.json

Reply:
@chicago-joe When you get a chance 🙂  ^